scaleway/resource_domain_record.go

```diff
@@ -277,9 +277,14 @@ func resourceScalewayDomainRecordCreate(ctx context.Context, d *schema.ResourceD
 		dnsZoneRes.Subdomain,
 		dnsZoneRes.Status))
 
+	name := d.Get("name").(string)
+	if name == "@" {
+		name = ""
+	}
+
 	dnsZoneData, err := domainAPI.ListDNSZoneRecords(&domain.ListDNSZoneRecordsRequest{
 		DNSZone: dnsZone,
-		Name:    d.Get("name").(string),
+		Name:    name,
 		Type:    recordType,
 	}, scw.WithAllPages(), scw.WithContext(ctx))
```

Comment:
Could it be done in an helper as this is done 2 times.
With a comment to explain the issue